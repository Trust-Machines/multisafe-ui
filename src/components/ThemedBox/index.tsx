import React from 'react';

import Box from '@mui/material/Box';
import {grey} from '@mui/material/colors';
import {useTheme} from '@mui/material';
import {SxProps} from '@mui/system';
import {Theme} from '@mui/material/styles';

const ThemedBox = (props: { children: React.ReactNode, sx?: SxProps<Theme> }) => {
    const theme = useTheme();

    return <Box sx={
        {
            ...{
                bgcolor: theme.palette.mode === 'light' ? '#fff' : grey[800],
                padding: '10px',
                borderRadius: '6px',
                boxShadow: `${theme.palette.divider} 1px 2px 10px 0px`
            },
            ...props.sx
        }
    }>{props.children}</Box>
}

export default ThemedBox