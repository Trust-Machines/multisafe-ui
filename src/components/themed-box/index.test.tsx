import {render} from '@testing-library/react';

import ThemedBox from './index';

test('1 Default render', () => {
    const rendered = render(<ThemedBox><span>children</span></ThemedBox>);
    expect(rendered.container).toMatchSnapshot()
});

test('2 Render with custom style', () => {
    const rendered = render(<ThemedBox sx={{background: 'red'}}><span>children</span></ThemedBox>);
    expect(rendered.container).toMatchSnapshot()
});

