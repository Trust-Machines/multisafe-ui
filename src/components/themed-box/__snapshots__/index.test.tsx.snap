// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1 Default render 1`] = `
<div>
  <div
    class="MuiBox-root css-14h3jy0"
  >
    <span>
      children
    </span>
  </div>
</div>
`;

exports[`2 Render with custom style 1`] = `
<div>
  <div
    class="MuiBox-root css-14k07pk"
  >
    <span>
      children
    </span>
  </div>
</div>
`;
